addons/mail/models/mail_activity.py

```diff
@@ -204,6 +204,8 @@ def _filter_access_rules(self, operation):
             valid = super(MailActivity, self)._filter_access_rules(operation)
             if valid and valid == self:
                 return self
+        elif operation == 'read':
+            valid = self.sudo().filtered_domain([('user_id', '=', self.env.uid)])
```

Comment:
Actually you can change 'mail_activity_rule_user' security rule and set 'perm_read' to True, and add this in the "'if" above (that way you rely more on acls, less on custom code)